### PR TITLE
pythonPackages.django-anymail: init at 6.1.0

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change
I'd like to build my project with Nix, and this package is needed.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/python-modules/django-anymail/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    requests
+    django
+  ];
+
+  doCheck = false;
+
+  # this package allows multiple email backends
+  # but each of those backends depends on a python package
+  # which are currently missing from the repositories
+  # only mailgun works by default
```
Comment (risicle) [inline pkgs/development/python-modules/django-anymail/default.nix]: We certainly have `boto3` which will enable SES.

Comment (ivegotasthma) [reply pkgs/development/python-modules/django-anymail/default.nix]: :+1:  thanks

File: pkgs/development/python-modules/django-anymail/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  propagatedBuildInputs = [
+    six
+    requests
+    django
+    boto3
+  ];
+
+  doCheck = false;
```
Comment (risicle) [inline pkgs/development/python-modules/django-anymail/default.nix]:
```
  src = fetchFromGitHub {
    owner = "anymail";
    repo = pname;
    rev = "v6.1";
    sha256 = "04jgz3qnsnba18rlqgxyb2g9128pk3ivflnj6695kibxg724fcpv";
  };
```

```
  checkInputs = [ mock ];
  checkPhase = ''
    substituteInPlace setup.py --replace "tests_require=[" "tests_require=[], #"
    export CONTINUOUS_INTEGRATION=1
    export ANYMAIL_SKIP_TESTS="sparkpost"
    ${python.interpreter} setup.py test
  '';
```

Comment (ivegotasthma) [reply pkgs/development/python-modules/django-anymail/default.nix]: dude, you're a magician. thanks for spending time on this and showing me how to do it in the future!

Comment (ivegotasthma) [reply pkgs/development/python-modules/django-anymail/default.nix]: in general, if I'm patching patching tests, wouldn't the `substituteInPlace` delete important requirement that break the tests?

Comment (risicle) [reply pkgs/development/python-modules/django-anymail/default.nix]: A trick you can often follow if you're having trouble with tests is to look at what their CI does. `.travis.yml` and `appveyor.yml` are a goldmine.

**Comment by ivegotasthma** [comment]
I think this is ready to be merged, is there something left @FRidh ?

**Pull request merged**